src/docs/arch-fast-truncate.dox

```diff
@@ -0,0 +1,843 @@
+utility.)
+
+In general the deletion information should only be consulted if the ref's state prior to
+locking was \c WT_REF_DELETED.
+Apart from instantiated pages, whose state is \c WT_REF_MEM (see @ref
+ft_instantiated_pages below) the \c page_del field will be \c NULL.
+
+Because locking the ref is expensive, and visibility checks are expensive as well, we want
+to avoid doing either unnecessarily.
+Consequently many of the places in the system that lock the ref to examine the deletion
+information will discard it right away if they find the deletion has become globally
+visible.
+
+Note that because on-disk pages can only be discarded at certain points, a page with null
+\c page_del may still have an on-disk page and consequently an address.
+That page presumptively contains no useful information.  (A page with non-null \c page_del
+should always have an on-disk page.)
+
+The transaction referenced in the deletion information may be committed or uncommitted,
+but it is never aborted.
+Upon transaction rollback the \c page_del field is cleared immediately.
+
+Note that it is not possible to delete a page twice; if the same region of a tree is
+truncated twice, the already-deleted pages will be skipped over the second time.
+See @ref ft_truncate.
+
+Also note that deleted pages still "exist" in the sense that the \c WT_REF remains, and
+searches that lead to its portion of the namespace will still land on it.
+(This triggers instantiation.)
+At some point after the deletion becomes globally visible, the \c WT_REF is discarded.
+This happens either during an internal page split or reverse split, or when the internal
+page is evicted.
+It is this discard process (rather than the deletion itself) that creates gaps in the
+namespace.
+
+@section ft_instantiated_pages Instantiated Pages
+
+An instantiated page is a page in \c WT_REF_MEM state that has been produced by the
+instantiation process (see @ref ft_instantiation).
+For most purposes instantiated pages are ordinary in-memory pages.
+
+Instantiated pages always have a modify structure, and differ from ordinary in-memory
+pages in two ways.
+First, the field \c page->modify->instantiated is set to \c true, and the \c page_del
+field is retained from the prior deleted state.
+Second, if the transaction that deleted the page had not resolved yet when the
+instantiation happened, the field \c page->modify->inst_updates contains an array of the
+\c WT_UPDATE structures used to do the instantiation.
+
+These two conditions are orthogonal and resolve independently; when both are resolved the
+page is thenceforth a completely normal in-memory page.
+
+Note that instantiated pages are not automatically marked dirty.
+(See the notes on this point in the @ref ft_instantiation section.)
+
+Also note that it is not possible to fast-truncate an instantiated page.
+Only on-disk pages can be fast-truncated.
+See @ref ft_truncate.
+
+@subsection ft_instantiated_instantiated Instantiated flag and \c page_del
```

Comment:
I generated the doc and it seems that `\c` does not get processed for headers. The same happens with the next header `inst_updates`

Reply:
Boo on doxygen :-(  Apparently you can't use the raw tt tag either. Fixed in [f3a555c](https://github.com/wiredtiger/wiredtiger/pull/8237/commits/f3a555cfd7ddc3655258d577632287a485c5676a)